docs(run-rails): updates for clarity and correctness (#1560)
NVIDIA/NeMo-Guardrails: docs/run-rails/generation-options.md

```json
{
  "role": "assistant",
  "content": "Some altered user input."
}
```

If the input was blocked, you will get the predefined response `bot refuse to respond` (by default "I'm sorry, I can't respond to that").

```json
{
  "role": "assistant",
  "content": "I'm sorry, I can't respond to that."
}
```

For more details on what rails was triggered, use the `log.activated_rails` generation option.

### Input and Output Rails Only

If you want to check both the user input and an output that was generated outside of the guardrails configuration, you must disable the dialog rails and the retrieval rails, and provide a bot message as well when making the call:

```python
res = rails.generate(messages=[{
    "role": "user",
    "content": "Some user input."
}, {
    "role": "assistant",
    "content": "Some bot output."
}], options={
    "rails": ["input", "output"]
})
```

The response will be the exact bot message provided, if allowed, an altered version if an output rail decides to change it, for example, to remove sensitive information, or the predefined message for `bot refuse to respond`, if the message was blocked.

> For more details on what rails was triggered, use the `log.activated_rails` generation option.

Reviewer: ```suggestion
For receive details on what rails are triggered, use the `log.activated_rails` generation option.
```

### Output Rails Only

> If you want to apply only the output rails to an LLM output, you must disable the input rails as well and provide an empty input.

Reviewer: ```suggestion
To apply output rails exclusively to an LLM response, disable the input rails and provide an empty input.
```

- [0.31s] OUTPUT (self check output): 1 actions (self_check_output), 1 llm calls [0.31s]
```

## Output Variables

Some rails can store additional information in [context variables](../configuration-guide/colang/colang-language-syntax-guide.md#variables). You can return the content of these variables by setting the `output_vars` generation option to the list of names for all the variables that you are interested in. If you want to return the complete context (this will also include some predefined variables), you can set `output_vars` to `True`.

```python
rails.generate(messages=messages, options={
    "output_vars": ["some_input_rail_score", "some_output_rail_score"]
})
```

> The returned data will be included in the `output_data` key of the response:

Reviewer: ```suggestion
You can find the returned data in the `output_data` key of the response:
```

```json
{
  "response": [...],
  "output_data": {
    "some_input_rail_score": 0.7,
    "some_output_rail_score": 0.8
  }
}
```

## Additional LLM Parameters

> You can pass additional parameters to the LLM call that is used to generate the final message by using the `llm_params` generation option. For example, to use a lower temperature than the default one:

Reviewer: ```suggestion
To supply additional parameters to the LLM call during final message generation, utilize the `llm_params` option. The following example demonstrates how to apply a lower value for `temperature`:
```

```python
rails.generate(messages=messages, options={
    "llm_params": {
        "temperature": 0.2
    }
})
```

> The supported parameters depend on the underlying LLM engine. The NeMo Guardrails library passes them "as is".

Reviewer: ```suggestion
The available parameters are determined by the specific LLM engine in use. The NeMo Guardrails library transmits values defined in the options parameter without modification.
```

## Additional LLM Output

> You can receive the additional output from the LLM generation by using the `llm_output` generation options.

Reviewer: ```suggestion
You can receive additional output from the LLM generation by setting `llm_output` to `True` through the `options` parameter.
```

```python
rails.generate(messages=messages, options={
    "llm_output": True
})
```

```{note}

> The data that is returned is highly dependent on the underlying implementation of the LangChain connector for the LLM provider. For example, for OpenAI, it only returns `token_usage` and `model_name`.

Reviewer: ```suggestion
The returned data is highly dependent on the underlying implementation of the LangChain connector for the LLM provider. For example, for OpenAI, it only returns `token_usage` and `model_name`.
```
